include/deal.II/cut/immersed_surface_quadrature.h

```diff
@@ -0,0 +1,84 @@
+#ifndef dealii__cut_immersed_surface_quadrature
+#define dealii__cut_immersed_surface_quadrature
+
+#include <deal.II/base/config.h>
+#include <deal.II/base/point.h>
+#include <deal.II/base/subscriptor.h>
+#include <deal.II/base/tensor.h>
+
+#include <vector>
+
+DEAL_II_NAMESPACE_OPEN
+namespace cut
+{
+
+  /**
+   * Defines a quadrature formula for integration over an oriented surface, $S$,
+   * immersed in the unit cell. By immersed it is meant that the surface may
+   * intersect the unit cell in an arbitrary way. The quadrature formula is
+   * described by a set of quadrature points and discrete surface elements,
+   * $\Delta\bar{S}$. The surface elements are normal to the surface and their
+   * norm equals the quadrature weights. Thus, the surface integral of a
+   * function, $f$, is approximated as
+   *
+   * $\int_S f dS \approx \sum_q f(x_q) \| \Delta \bar{S}_q \|$.
+   *
+   */
+  template<int dim>
+  class ImmersedSurfaceQuadrature : public Subscriptor
+  {
+  public:
+
+    /**
+     * Default constructor initialize the quadrature with no quadrature points.
+     */
+    ImmersedSurfaceQuadrature();
+
+    /**
+     * Extend the given formula by an additional quadrature point. This function
+     * should only be used during construction of the quadrature formula.
+     */
+    void add_point(const Point<dim> &point,
+                   const Tensor<1, dim> &surface_element);
+
+    /**
+     * Return a reference to the <tt>i</tt>th surface element.
+     */
+    const Tensor<1,dim> &surface_element(const unsigned int i) const;
+
+    /**
+     * Return a reference to the <tt>i</tt>th quadrature point.
+     */
+    const Point<dim> &point(const unsigned int i) const;
+
+    /**
+     * Return a reference to the whole %vector of quadrature points.
+     */
+    const std::vector<Point<dim> > &get_points() const;
```

Comment:
we now require C++11 (gcc4.8 and alike), so you don't need extra space in `> >`. Same below.